Add random_name_suffix to SparkKubernetesOperator (#43800)

mrk-andreev opened the pull request:
Prior to this change, `random_name_suffix` was only documented but not implemented as a configurable option. Passing this value as an argument had no effect. This commit introduces a `false` option for `random_name_suffix`, which prevents the generation of a random suffix for the pod name. For compatibility, the default value is set to `true`, ensuring the pod name will still conform to `MAX_LABEL_LEN = 63`.

Fixes: #43800

mrk-andreev committed:
Add random_name_suffix to SparkKubernetesOperator (#43800)

Prior to this change, `random_name_suffix` was only documented but not implemented as a configurable option. Passing this value as an argument had no effect. This commit introduces a `false` option for `random_name_suffix`, which prevents the generation of a random suffix for the pod name. For compatibility, the default value is set to `true`, ensuring the pod name will still conform to `MAX_LABEL_LEN = 63`.

Fixes: #43800

potiuk commented on providers/tests/cncf/kubernetes/operators/test_spark_kubernetes.py:
```diff
@@ -828,3 +830,26 @@ def test_resolve_application_file_real_file_not_exists(create_task_instance_of_o
+    pod_name = op.create_job_name()
+
+    if random_name_suffix:
+        assert pod_name.startswith(name)
```
```suggestion
        assert pod_name.startswith(name)
```

we could also check if the pod_name is different than name

mrk-andreev replied:
Withed with 

```
assert pod_name.startswith(name)
assert pod_name != name
```

mrk-andreev committed:
Add random_name_suffix to SparkKubernetesOperator (#43800)

Prior to this change, `random_name_suffix` was only documented but not implemented as a configurable option. Passing this value as an argument had no effect. This commit introduces a `false` option for `random_name_suffix`, which prevents the generation of a random suffix for the pod name. For compatibility, the default value is set to `true`, ensuring the pod name will still conform to `MAX_LABEL_LEN = 63`.

Fixes: #43800
